### PR TITLE
Add support for organizationNameKey for oauth token exchange

### LINKED ISSUE #742: Add support for organiationNameKey for oauth token exchange
https://stormpath.atlassian.net/browse/AM-3287

### PR DESCRIPTION
**Pull request by jarias**
Fixes #742

### CODE REVIEW

File: extensions/httpclient/src/test/groovy/com/stormpath/sdk/impl/oauth/OAuthPasswordGrantRequestAuthenticatorIT.groovy [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package com.stormpath.sdk.impl.oauth
```
Comment (lhazlewood) [inline extensions/httpclient/src/test/groovy/com/stormpath/sdk/impl/oauth/OAuthPasswordGrantRequestAuthenticatorIT.groovy]: header

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/DefaultWrappedServletRequestFactory.java [hunk 2]
```diff
@@ -34,24 +35,28 @@
...
                                                Saver<AuthenticationResult> authenticationResultSaver,
                                                Publisher<RequestEvent> eventPublisher,
+                                               Resolver<String> organizationNameKeyResolver,
                                                String userPrincipalStrategyName, String remoteUserStrategyName) {
         this.usernamePasswordRequestFactory = factory;
         this.authenticationResultSaver = authenticationResultSaver;
         this.eventPublisher = eventPublisher;
         this.userPrincipalStrategyName = userPrincipalStrategyName;
         this.remoteUserStrategyName = remoteUserStrategyName;
+        this.organizationNameKeyResolver = organizationNameKeyResolver;
```
Comment (lhazlewood) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/DefaultWrappedServletRequestFactory.java]: I may not understand the scope of this, but I don't think the request should know anything about org name keys - just the account store that is relevant for the request (obtained by an `AccountStoreResolver`).

In other words, I think the AccountStoreResolver is the primary component and it in turn delegates to an OrganizationNameKeyResolver to look up an org name key to use when finding the corresponding Organization (i.e. AccountStore).

Does this make sense?

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/http/impl/StormpathHttpServletRequest.java [hunk 3]
```diff
@@ -101,11 +102,13 @@
     private final Publisher<RequestEvent> eventPublisher;
     private final String userPrincipalStrategyName;
     private final String remoteUserStrategyName;
+    private final Resolver<String> organizationNameKeyResolver;
```
Comment (lhazlewood) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/http/impl/StormpathHttpServletRequest.java]: Same - probably want to reference the relevant AccountStore, not the resolver, no?

Comment (lhazlewood) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/http/impl/StormpathHttpServletRequest.java]: Also, I'd assume that the org (account store) would be resolved early in the request, presumably via a servlet filter, and once discovered, set on the request (i.e. `request.setAttribute("accountStore", organization);` )

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/DefaultWrappedServletRequestFactory.java (hunk 2 above)
Comment (jarias) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/DefaultWrappedServletRequestFactory.java]: Makes sense. I'll check if the AccountStoreResolver really delagates to OrganizationNameKeyResolver

Comment (jarias) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/DefaultWrappedServletRequestFactory.java]: But does the API token endpoint checks if the account store is an org and sets the org claim in the resulting access token?

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/http/impl/StormpathHttpServletRequest.java (hunk 3 above)
Comment (jarias) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/http/impl/StormpathHttpServletRequest.java]: Looks like we don't have a valid impl of an account store resolver nor a filter, I think @mrioan was working on this for ticket https://github.com/stormpath/stormpath-sdk-java/issues/351 may this needs to be finish before we can wrap this one out.

**Pull request closed without merging**